v2/src/lib.rs

```diff
@@ -0,0 +1,837 @@
+    if !dbs.is_empty() {
+        Ok(true)
+    } else {
+        Ok(false)
+    }
+}
+
+//this should return the DB_ID
+async fn create_db(client: &Client, org_id: &str, db: &str) -> ZResult<bool> {
+    //let mut db_id: String = String::new();
+    let result = client
+        .create_bucket(Some(PostBucketRequest::new(
+            org_id.to_owned(),
+            db.to_owned(),
+        )))
+        .await;
+    match result {
+        Ok(_) => Ok(true),
+        Err(_) => Ok(false), //can post error here
+    }
+}
+// Returns an InfluxDB regex (see https://docs.influxdata.com/influxdb/v1.8/query_language/explore-data/#regular-expressions)
+// corresponding to the list of path expressions. I.e.:
+// Replace "**" with ".*", "*" with "[^\/]*"  and "/" with "\/".
+// Concat each with "|", and surround the result with '/^' and '$/'.
+fn key_exprs_to_influx_regex(path_exprs: &[&keyexpr]) -> String {
+    let mut result = String::with_capacity(2 * path_exprs[0].len());
+    result.push_str("/^");
+    for (i, path_expr) in path_exprs.iter().enumerate() {
+        if i != 0 {
+            result.push('|');
+        }
+        let mut chars = path_expr.chars().peekable();
+        while let Some(c) = chars.next() {
+            match c {
+                '*' => {
+                    if let Some(c2) = chars.peek() {
+                        if c2 == &'*' {
+                            result.push_str(".*");
+                            chars.next();
+                        } else {
+                            result.push_str(".*")
+                        }
+                    }
+                }
+                '/' => result.push_str(r"\/"),
+                _ => result.push(c),
+            }
+        }
+    }
+    result.push_str("$/");
+    result
+}
+
+fn time_from_parameters(t: &str) -> ZResult<Option<(u64, u64)>> {
+    use zenoh::selector::{TimeBound, TimeRange};
+    let time_range = t.time_range()?;
+
+    let mut start_time: u64 = 0;
+    let mut stop_time: u64 = Utc::now().naive_utc().timestamp().try_into().unwrap();
+
+    match time_range {
+        Some(TimeRange(start, stop)) => {
+            match start {
+                TimeBound::Inclusive(t) => {
+                    start_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Exclusive(t) => {
+                    start_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Unbounded => {}
+            }
+            match stop {
+                TimeBound::Inclusive(t) => {
+                    stop_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Exclusive(t) => {
+                    stop_time = calculate_time(t).unwrap();
+                }
+                TimeBound::Unbounded => {}
+            }
+        }
+        None => {
+            return Ok(None);
+        }
+    }
+    Ok(Some((start_time, stop_time)))
+}
+
+fn calculate_time(tx: TimeExpr) -> ZResult<u64> {
```

Comment:
The time calculated here is in second, and thus truncating sub-second parts which can be important in some use cases.
InfluxDB v1 .x was supporting sub-second time expressions, either as a float (in seconds), either as RFC-3339 format. I would be surprise InfluxDB v2.x doesn't.

Reply:
Done.